packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1227,7 +1227,7 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
     }
   }
 
-  TextStyle? get _textStyle => widget.style ?? Theme.of(context).textTheme.subtitle1;
+  TextStyle? get _textStyle => Theme.of(context).textTheme.subtitle1?.merge(widget.style);
```

Comment:
Not sure this is what we want to do here. If the app asked for a specific text style, why would we want to always merge it with the `subtitle1` style?  The previous code was just using subtitle1 as a default style in case the app didn't provide one.

Reply:
My bad, I wasn't clear with the issue description
When trying to reproduce the issue using the sample code from https://github.com/flutter/flutter/issues/68004#issue-720030578
I noticed custom text style is always white in the light theme and white in the dark theme as expected (initially tested on mobile) but now testing I can see it works on the web but not on mobile or desktop. 
White in the light theme makes it hard to read, this issue occurs when text style just font size, when color is not provided. Other widgets keep text color the same when providing custom text style with just font size. 

Here I cleaned up the example provided by the author of the issue

<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

/// This is the main application widget.
class MyApp extends StatelessWidget {
  static const String _title = 'Flutter Code Sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(title: const Text(_title)),
        body: Center(
          child: MyStatefulWidget(),
        ),
      ),
    );
  }
}

/// This is the stateful widget that the main application instantiates.
class MyStatefulWidget extends StatefulWidget {
  MyStatefulWidget({Key? key}) : super(key: key);

  @override
  _MyStatefulWidgetState createState() => _MyStatefulWidgetState();
}

/// This is the private State class that goes with MyStatefulWidget.
class _MyStatefulWidgetState extends State<MyStatefulWidget> {
  String dropdownValue = 'One';

  @override
  Widget build(BuildContext context) {
    return DropdownButton<String>(
      value: dropdownValue,
      //todo uncomment line below to test ------------
      style: TextStyle(fontSize: 14),
      
      onChanged: (String? newValue) {
    
      },
      items: <String>['One', 'Two', 'Free', 'Four']
          .map<DropdownMenuItem<String>>((String value) {
        return DropdownMenuItem<String>(
          value: value,
          child: Text(value),
        );
      }).toList(),
    );
  }
}
``` 
	
</details>


### Code snippet
```dart
    return DropdownButton<String>(
      value: dropdownValue,
      //todo uncomment line below to test ------------
      style: TextStyle(fontSize: 14),
      onChanged: (String? newValue) {
      },
      items: <String>['One', 'Two', 'Free', 'Four']
          .map<DropdownMenuItem<String>>((String value) {
        return DropdownMenuItem<String>(
          value: value,
          child: Text(value),
        );
      }).toList(),
    );
```

| Desktop | Desktop |
| --------------- | --------------- |
| <img src="https://user-images.githubusercontent.com/48603081/148971448-f7d86e06-91e7-4546-a88c-e6b1a03eb2bd.png"  /> | <img src="https://user-images.githubusercontent.com/48603081/148971458-934a925d-1106-4354-b676-e479c96c3768.png" /> |

| Mobile | Web |
| --------------- | --------------- |
| <img src="https://user-images.githubusercontent.com/48603081/148971593-f1509055-74b7-476f-9f43-5311f75ed945.png" height="450" /> |  <img src="https://user-images.githubusercontent.com/48603081/148971951-fde78d0d-bee5-4129-bae7-1a0081e53999.png" height="450" />	|

I just updated the PR to merge DefaultTextStyle to fix issue similar to `selectable_text.dart`
https://github.com/flutter/flutter/blob/9c5b390710bd72e3c5d7e942a70599c2e9d8c78b/packages/flutter/lib/src/material/selectable_text.dart#L656-L660


With the changes 
Now it renders the text properly 
| Preview | 
| --------------- | 
| <img src="https://user-images.githubusercontent.com/48603081/148973347-cf2d4edc-17a5-4431-b442-3e16c6292626.png" height="450" />	|

Reply:
cc: @darrenaustin